pkgs/applications/misc/cura/lulzbot/curaengine.nix

```diff
@@ -10,6 +10,8 @@ gcc8Stdenv.mkDerivation rec {
     sha256 = "0wdkvg1hmqp1gaym804lw09x4ngf5ffasd861jhflpy7djbmkfn8";
   };
 
+  patches = [ ./curaengine-openmp-compat.patch ];
```

Comment:
This patch and its content definitely need some comments.

* Where is it from, who wrote it?
* Is there any plan/issue about upstreaming?
* Why are these openmp ptragmas more correct than what was there before?

Reply:
- The patch is purely my own work.
- Upstream has been contacted -- I'm a Lulzbot customer with an active support contract and sent them the patch. That said, they're in the middle of some major corporate churn (sold themselves out of near-bankruptcy to an out-of-state business entity formed as a holding company; moved to that state; have been slowly restaffing after), so a response may take a while.
- The new version of OpenMP does not allow outside variables to be referenced *at all* without an explicit declaration of how they're supposed to be handled. Thus, this was an outright build failure beforehand. The new pragmas copy the initial value from the outer scope into each parallel thread. Since these variables are all constant within the loops, this is clearly correct.

Reply:
...will be amending with comments conveying the same information as the above.

Reply:
Sounds good to me.

Reply:
That comment is present in the current version of the patch (as [pushed last Sunday](https://github.com/NixOS/nixpkgs/commit/68d14b3756629fd409334d7b8b615c8ecd26cd4a)).